### PR TITLE
Improvements to per angle options

### LINKED ISSUE #22039: Minor improvements to per-angle options
There are a few minor issues with handling per-angle options in the ISIS Reflectometry GUI in the 3.12 beta:

- For the new GUI to be useful on SURF and CRISP, scientists require that additional options are available on a per-angle basis on the Settings tab: Q min, Q max, dQ/Q and Scale.
- the Get Defaults button on the Settings tab clears values for which a default doesn't exist. If a default doesn't exist then any entered values should remain unchanged.
- Duplicate angles in the table are ignored and the last one given takes precedence. It would be slightly clearer if the first one took precedence instead.

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR adds the following changes to the ISIS Reflectometry GUI:

- Add additional options that can be set on a per-angle basis on the Settings tab: Q min, Q max, dQ/Q and Scale.
- Fix a bug where the Get Defaults button on the Settings tab clears values for which a default doesn't exist
- If duplicate angles are given, the first one takes precedence

The large number of files changed is due to renaming a pass-through funtion.

**To test:**
Test default options:
- Open the `ISIS Reflectometry` GUI
- Set the instrument to `INTER` and enter a Run number `13460`
- On the Settings tab, in the table enter some values, e.g.
  - Transmission runs = `13463,13464`
  - Q min = 0.008
  - Q max = 0.1
  - dQ/Q = 0.02
  - Scale = 0.9
- On the Runs tab, click Process
- Check that all the values are populated with the same values in the table on the Runs tab (apart from transmission runs - historically this was never populated)
- Check that a workspace `TRANS_13463+13464` has been created. This indicates that the correct transmission runs were used.

Test per-angle lookup:
- Delete all workspaces, and clear all values in the Runs tab table apart from the run number and the angle. 
- On the Settings tab, leave the values as before but add `Angle = 0.7`
- Add another row with the same angle and different values.
- Add another row with a different angle and different values.
- Re-process and check you get the same results. This checks that the angle lookup found the correct row and ignored the second duplicate angle.

Test defaults:
- On the settings tab, click Get Defaults. The values in the table should remain unchanged because they do not have values in the INTER parameters file.

Fixes #22039 

**Release Notes** 
*Updated release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by edwardb96 — CHANGES_REQUESTED** [review]
Working with the exception of the bug I described to you. Some code-style suggestions...

File: qt/scientific_interfaces/ISISReflectometry/ReflSettingsPresenter.cpp [hunk 1]
```diff
@@ -226,8 +223,10 @@ OptionsQMap ReflSettingsPresenter::getReductionOptions() const {
     // provide per-angle transmission runs, but these need to be
     // requested on a per-row basis when we know what the angle is
     // for that row.
-    auto transmissionRuns = getDefaultTransmissionRuns();
-    addIfNotEmpty(options, "FirstTransmissionRun", transmissionRuns);
+    auto defaultOptions = getDefaultOptions();
+    for (auto iter = defaultOptions.begin(); iter != defaultOptions.end();
```
[the remark this answers is not in the document]
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/ReflSettingsPresenter.cpp]: Yes, but I need keys as well as values and a range-based for loop with a QMap just seems to give the values

File: qt/scientific_interfaces/ISISReflectometry/QtReflSettingsView.cpp [hunk 2]
```diff
@@ -139,18 +146,14 @@ void QtReflSettingsView::registerExperimentSettingsWidgets(
     Mantid::API::IAlgorithm_sptr alg) {
   connectSettingsChange(*m_ui.expSettingsGroup);
   registerSettingWidget(*m_ui.analysisModeComboBox, "AnalysisMode", alg);
-  registerSettingWidget(*m_ui.transmissionRunsTable, "FirstTransmissionRun",
-                        alg);
+  registerSettingWidget(*m_ui.optionsTable, "FirstTransmissionRun", alg);
```
[the remark this answers is not in the document]
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/QtReflSettingsView.cpp]: Doh, yes!

**Review by edwardb96 — APPROVED** [review]

**Comment by edwardb96** [comment]
Ok to merge when builds pass. :+1:

**Pull request merged**